sycl/test/on-device/online_compiler/online_compiler_common.hpp

```diff
@@ -0,0 +1,171 @@
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/online_compiler.hpp>
+
+#include <iostream>
+#include <vector>
+
+static const char *CLSource = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+)===";
+
+static const char *CLSourceSyntaxError = R"===(
+__kernel void my_kernel(__global int *in, __global int *out) {
+  syntax error here
+  size_t i = get_global_id(0);
+  out[i] = in[i]*2 + 100;
+}
+)===";
+
+static const char *CMSource = R"===(
+extern "C"
+void cm_kernel() {
+}
+)===";
+
+using namespace sycl::INTEL;
+
+void testSyclKernel(sycl::queue &Q, sycl::kernel Kernel) {
+  std::cout << "Run the kernel now:\n";
+  const int N = 4;
+  int InputArray[N] = {0, 1, 2, 3};
+  int OutputArray[N] = {};
+
+  sycl::buffer<int, 1> InputBuf(InputArray, sycl::range<1>(N));
+  sycl::buffer<int, 1> OutputBuf(OutputArray, sycl::range<1>(N));
+
+  Q.submit([&](sycl::handler &CGH) {
+     CGH.set_arg(0, InputBuf.get_access<sycl::access::mode::read>(CGH));
+     CGH.set_arg(1, OutputBuf.get_access<sycl::access::mode::write>(CGH));
+     CGH.parallel_for(sycl::range<1>{N}, Kernel);
+   }).wait();
+
+  auto Out = OutputBuf.get_access<sycl::access::mode::read>();
+  for (int I = 0; I < N; I++) {
+    std::cout << I << "*2 + 100 = " << Out[I] << "\n";
+  }
+}
+
+template <cl::sycl::INTEL::source_language Lang>
+void doCompileAndRunTest(const std::string &Source) {
+  online_compiler<Lang> Compiler;
+}
+
+int main(int argc, char **argv) {
+  cl::sycl::queue Q;
+  cl::sycl::context Context = Q.get_context();
+  cl::sycl::device Device = Q.get_device();
+
+  { // Compile and run a trivial OpenCL kernel.
+    std::cout << "Test case1\n";
+    online_compiler<source_language::opencl_c> Compiler;
+    std::vector<byte> IL;
+    try {
+      IL = Compiler.compile(
+          std::string(CLSource),
+          // Intentionally use one option twice.
+          std::vector<std::string>{std::string("-cl-fast-relaxed-math"),
```

Comment:
Duplicate option

Reply:
It was just a harmless test that checked that more than 1 argument is accepted. I will replace the 2nd arg with a different one.

Reply:
yeah, different one looks safer. ocloc might start complaining about duplicate options

Reply:
Fixed in 392cc468cb3b1d41a23c32b9546339c0b3c19057